ui/src/plugins/Notify.json

```diff
@@ -390,150 +418,10 @@
         "params": {
           "props": {
             "type": "Object",
+            "tsType": "QNotifyUpdateOptions",
             "required": false,
-            "desc": "Notification properties that will be shallow merged to previous ones in order to update the non-grouped notification",
```

Comment:
Sorry for not pointing this earlier, but we shouldn't remove these as well, even though there is a tsType defined now.

JSONs are not only for helping TS defs. They are consumed as standalone too. Example 1: docs API card; Example 2: quasar describe command.

Reply:
That's why I added "reference"/"pointer" messages like the following:
> See 'opts' param of 'create()' for object properties, except 'group' and 'position'

We are already using this approach in a lot of places. One example is that in `actions`, we are pointing out `QBtn` props, because 1. it's easy to get lost as there are lots of nested properties, 2. it's really hard to keep JSON API definitions in sync because we can't re-use whole structures, unlike TS. Checking out the API card for `Notify.create` is really hard, because there are lots of properties in parameters, and there are lots more in the return type as well, most are the same, hard to spot the difference by eye. Definitions, descriptions, etc. were already out-of-sync in this file. So, a simple explanation and a pointer are more helpful and simple IMO, both for users and maintainers. But, if you still think we should add it, I'll add them back.

Reply:
Missed the "see opts param". It's not a perfect solution but it's pretty good because besides the maintenance benefit it also decreases the size of the JSON file -- and I'm constantly researching ways to decrease the footprint of the docs (as some users that don't use the PWA are complaining about high load times).

As for the docs API cards, I'm planning some tweaking to improve the UX.

Maybe some thoughts for force keeping defs in sync: we do an AST analysis to catch possible problems, but it only goes so far. Once a composable is used the analysis does not cover it (same for any other imported file). For someone who has a bit of time to enhance the AST analysis this would be something really great for the framework. Then we should do something about the TS defs which are manually defined so as to not get out of sync with UI code and JSON.

Reply:
I don't have a lot of ideas about optimizing the JSON files. I will keep this problem in mind to see if I can think of anything.

UX improvements sound great.

For keeping things in sync: I was thinking of a TS-first approach(_and experimenting with it_). As TypeScript + JSDoc/TSDoc is more detailed and comprehensive than JSON API, we can auto-generate JSON API from TypeScript. We have some extra metadata such as `category`, `addedIn`, etc. in JSON API, some of them are available in JSDoc/TSDoc specs, some of them can be easily added as custom tags. We still need to preserve&support JSON API because we are using it in our docs, the describe command, in AEs, etc. Furthermore, since we will use JSON API as the common API, we can progressively migrate and enhance types. So, we don't have to convert all the types at once. I am also thinking of writing a script to convert an existing JSON API file to a separate `.d.ts` in the related component/composable/etc. folder, to use it as a base. It's easy because currently, we can convert JSON -> TS too.

By adopting this, we can provide generic types(_e.g. `QSelect<TModel, TOption>`_), have more detailed types, have an easier time when creating/syncing types, thus reducing the maintenance costs, etc.

A short example of what I have(_and aiming to have_) after a limited time of working on this idea:
```ts
// /ui/src/components/select/QSelect.d.ts

/**
 * @public
 * @docsUrl https://v2.quasar.dev/vue-components/select
 */
export interface QSelectProps<Model = any, Option = QSelectOption<Model>>
  extends UseFieldProps<Model>,
    UseVirtualScrollProps<QSelect> {
  /**
   * Model of the component.
   * Must be Array if using 'multiple' prop.
   * Either use this property (along with a listener for 'update:modelValue' event) OR use v-model directive
   *
   * @category model
   * @see https://v2.quasar.dev/vue-components/select#the-model
   * @example
   * ```
   * v-model="myModel"
   * ```
   */
  modelValue: Model;

  /**
   * Available options that the user can select from.
   * For best performance freeze the list of options.
   *
   * @category options
   * @defaultValue `[]`
   * @example String options
   * ```
   * :options="[ 'BMW', 'Samsung Phone' ]"
   * ```
   * @example Object options
   * ```
   * :options="[ { label: 'BMW', value: 'car' }, { label: 'Samsung Phone', value: 'phone' } ]"
   * ```
   */
  options?: Option[];

  /**
   * Property of option which holds the 'value'.
   * If using a function then for best performance, reference it from your scope and do not define it inline.
   *
   * @category options
   * @defaultValue 'value'
   * @param option The current option being processed.
   * @returns Value of the current option.
   * @example Property name form
   * ```
   * option-value="modelNumber"
   * ```
   * @example Function form
   * ```
   * :option-value="(item) => item === null ? null : item.modelNumber"
   * ```
   */
  optionValue?: FnOrKey<Option, Model>;

// ...

// /ui/src/composables/private/use-field.d.ts
export interface UseFieldProps<T = any> extends UseValidateProps<T> {
  /**
   * Used to specify the name of the control.
   * Useful if dealing with forms.
   * If not specified, it takes the value of 'for' prop, if it exists.
   */
  name?: string;
// ...
```
We can also reference these types in JS source code in form of JSDocs, and improve the DX in our source code too. Should we plan to migrate our code-base to TS, we can use these types. So, I see a lot of value in this. Please let me know what you think.

Reply:
Sounds really juicy. Let me think about it a bit and will come back to you shortly.